### ISSUE TITLE
mopidy container keeps restarting

### ISSUE THREAD

**Issue by logue** [RoggerFabri/mopidy-icesnap#27]
I am using [rfabri/mopidy:arm64v8-latest](https://hub.docker.com/layers/rfabri/mopidy/arm64v8-latest/images/sha256-0d86d4b395346494cbccc159dbc51307fbe3bcf3734a5565c67cf3e4e1266d2b?context=explore) of mopidy's container, but I get the following error and can restart repeatedly.

```
usage: mopidy [-h] [--version] [-q] [-v] [--config FILES] [-o OPTIONS]
unrecognized command: mopidy
```

I can't even get into a shell inside the container.

**Comment by RoggerFabri** [comment]
@logue can you please provide a docker compose or docker run command you're using?

**Comment by logue** [comment]
I am using [portainer](https://www.portainer.io/). Internally, I think that the following yaml is just started with `docker-compose up`.

```yml
version: "2.4"

services:
  icecast:
    container_name: icecast
    image: rfabri/icecast:amd64-latest
    restart: unless-stopped
    environment:
      - ICECAST_SOURCE_PASSWORD=mopidy
      - ICECAST_ADMIN_PASSWORD=mopidy
      - ICECAST_PASSWORD=mopidy
      - ICECAST_RELAY_PASSWORD=mopidy
    ports:
      - "0.0.0.0:8000:8000"

  mopidy-icecast:
    container_name: mopidy-icecast
    image: rfabri/mopidy:amd64-latest
    restart: unless-stopped
    command: mopidy
    ports:
      - "0.0.0.0:6601:6600"
      - "0.0.0.0:6681:6680"
    depends_on:
      - icecast
    volumes:
      - /volume1/Docker/mopidy-icesnap/mopidy/mopidy-icecast.conf:/mopidy.conf
      - /volume1/Docker/mopidy-icesnap/:/mopidy
```

**Comment by RoggerFabri** [comment]
Did you make a mistake somewhere, maybe? On your first post you said you're using `rfabri/mopidy:arm64v8-latest` but on your docker-compose you are using `rfabri/mopidy:amd64-latest`, those are two different images for two different architectures.

What kind of host you're running this in? Is the container `mopidy-icecast` running? Do you get any logs? What's the output of `docker logs mopidy-snapserver`?

**Comment by logue** [comment]
Sorry, Both are using `amd64-latest`.

Of these, the `icecast` side is working properly, and when I streamed from another PC with the icecast client, I was able to send it to the destination without any problems.

It's only `mopidy-icecast` that has issues.

I checked the log in the shell and got the same error. `unrecognized command: mopidy` follows.

```sh
$ sudo docker logs mopidy-icecast

usage: mopidy [-h] [--version] [-q] [-v] [--config FILES] [-o OPTIONS]

unrecognized command: mopidy
usage: mopidy [-h] [--version] [-q] [-v] [--config FILES] [-o OPTIONS]

unrecognized command: mopidy

...(omit)...

usage: mopidy [-h] [--version] [-q] [-v] [--config FILES] [-o OPTIONS]

unrecognized command: mopidy
```

**Comment by RoggerFabri** [comment]
@logue can you please remove the line below from the `mopidy-icecast` block?

`command: mopidy`

```
  mopidy-icecast:
    container_name: mopidy-icecast
    image: rfabri/mopidy:amd64-latest
    restart: unless-stopped
    command: mopidy   #REMOVE THIS LINE
    ports:
      - "0.0.0.0:6601:6600"
      - "0.0.0.0:6681:6680"
    depends_on:
      - icecast
    volumes:
      - /volume1/Docker/mopidy-icesnap/mopidy/mopidy-icecast.conf:/mopidy.conf
      - /volume1/Docker/mopidy-icesnap/:/mopidy
```

This is from older images and shouldn't be there any more. I'll update the docker-compose examples to reflect that.

**Comment by logue** [comment]
When I removed `command: mopidy` the log was:

```shell
INFO     2023-04-27 17:12:40,327 [1:MainThread] mopidy.__main__
  Starting Mopidy 3.4.1
INFO     2023-04-27 17:12:40,501 [1:MainThread] mopidy.config
  Loading config from builtin defaults
INFO     2023-04-27 17:12:40,504 [1:MainThread] mopidy.config
  Loading config from command line options
INFO     2023-04-27 17:12:40,549 [1:MainThread] mopidy.internal.path
  Creating dir file:///root/.cache/mopidy
INFO     2023-04-27 17:12:40,549 [1:MainThread] mopidy.internal.path
  Creating dir file:///root/.config/mopidy
INFO     2023-04-27 17:12:40,549 [1:MainThread] mopidy.internal.path
  Creating dir file:///root/.local/share/mopidy
INFO     2023-04-27 17:12:40,566 [1:MainThread] mopidy.config
  Loading config from builtin defaults
INFO     2023-04-27 17:12:40,569 [1:MainThread] mopidy.config
  Loading config from command line options
INFO     2023-04-27 17:12:40,578 [1:MainThread] mopidy.internal.path
  Creating file file:///mopidy/mopidy.conf
INFO     2023-04-27 17:12:40,579 [1:MainThread] mopidy.__main__
  Initialized file:///mopidy/mopidy.conf with default config
INFO     2023-04-27 17:12:41,184 [1:MainThread] mopidy.__main__
  Enabled extensions: softwaremixer, mpd, mobile, file, stream, http, iris, m3u, tunein
INFO     2023-04-27 17:12:41,184 [1:MainThread] mopidy.__main__
  Disabled extensions: local, scrobbler
WARNING  2023-04-27 17:12:41,184 [1:MainThread] mopidy.__main__
  Found local configuration errors. The extension has been automatically disabled:
WARNING  2023-04-27 17:12:41,184 [1:MainThread] mopidy.__main__
    local/media_dir must be set.
WARNING  2023-04-27 17:12:41,184 [1:MainThread] mopidy.__main__
  Found scrobbler configuration errors. The extension has been automatically disabled:
WARNING  2023-04-27 17:12:41,184 [1:MainThread] mopidy.__main__
    scrobbler/username must be set.
WARNING  2023-04-27 17:12:41,184 [1:MainThread] mopidy.__main__
    scrobbler/password must be set.
WARNING  2023-04-27 17:12:41,185 [1:MainThread] mopidy.__main__
  Please fix the extension configuration errors or disable the extensions to silence these messages.
INFO     2023-04-27 17:12:41,487 [1:MainThread] mopidy.commands
  Starting Mopidy mixer: SoftwareMixer
INFO     2023-04-27 17:12:41,489 [1:MainThread] mopidy.commands
  Starting Mopidy audio
INFO     2023-04-27 17:12:41,493 [1:MainThread] mopidy.commands
  Starting Mopidy backends: TuneInBackend, FileBackend, M3UBackend, StreamBackend
INFO     2023-04-27 17:12:41,504 [1:MainThread] mopidy.internal.path
  Creating dir file:///root/.local/share/mopidy/m3u
INFO     2023-04-27 17:12:41,509 [1:Audio-2] mopidy.audio.actor
  Audio output set to "autoaudiosink"
INFO     2023-04-27 17:12:41,515 [1:MainThread] mopidy.commands
  Starting Mopidy core
INFO     2023-04-27 17:12:41,527 [1:MainThread] mopidy.commands
  Starting Mopidy frontends: MpdFrontend, IrisFrontend, HttpFrontend
INFO     2023-04-27 17:12:41,529 [1:MainThread] mopidy_mpd.actor
  MPD server running at [::ffff:127.0.0.1]:6600
INFO     2023-04-27 17:12:41,530 [1:IrisFrontend-9] mopidy_iris.core
  Starting Iris 3.66.1
INFO     2023-04-27 17:12:41,532 [1:IrisFrontend-9] mopidy.internal.path
  Creating dir file:///root/.local/share/mopidy/iris
INFO     2023-04-27 17:12:41,532 [1:HttpFrontend-11] mopidy.http.actor
  HTTP server running at [::ffff:127.0.0.1]:6680
INFO     2023-04-27 17:12:41,534 [1:MainThread] mopidy.commands
  Starting GLib mainloop
INFO     2023-04-27 17:12:41,544 [1:HttpServer] mopidy.internal.path
  Creating dir file:///root/.local/share/mopidy/http
```

Since it was invisible from the browser, I tried to connect with wget in the host shell, but an error `Connection reset by peer` occurred.


> If you really want to provide a command, please use something like:
> 
> ```
> command: >
>       "mopidy --config /mopidy/mopidy.conf"
> ```
> 
> But given your docker-compose I don't think you need it.

This method was useless because an error ｀unrecognized command: mopidy` occurred.

**Comment by RoggerFabri** [comment]
How does your `mopidy.conf` looks like? From your logs, Mopidy is running and Iris should be accessible on port 6681, both Iris and Mopidy are running successfully.

Iris is running here:
```
INFO     2023-04-27 17:12:41,532 [1:HttpFrontend-11] mopidy.http.actor
  HTTP server running at [::ffff:127.0.0.1]:6680
```

**Comment by RoggerFabri** [comment]
Your mopidy.conf has to have http enabled like in here: https://docs.mopidy.com/en/latest/ext/http/#configuration

Something along those lines:

```
[mpd]
hostname = 0.0.0.0

[http]
enabled = true
hostname = 0.0.0.0
default_app = iris
```

Looks like you haven't set any config on your `mopidy.conf` and because of that the default configuration was used:

```
INFO     2023-04-27 17:12:40,578 [1:MainThread] mopidy.internal.path
  Creating file file:///mopidy/mopidy.conf
INFO     2023-04-27 17:12:40,579 [1:MainThread] mopidy.__main__
  Initialized file:///mopidy/mopidy.conf with default config
```
If I'm not mistaken, http is disabled by default.

**Comment by RoggerFabri** [comment]
Have you mounted these volumes to actual files on your host, or are you using volume mounts?

```
...
    volumes:
      - /volume1/Docker/mopidy-icesnap/mopidy/mopidy-icecast.conf:/mopidy.conf
      - /volume1/Docker/mopidy-icesnap/:/mopidy
```

I suggest you mount this to an existing path within your host, so you can edit as you wish, it will make it easier for you to debug.

**Comment by logue** [comment]
It worked when I changed the path setting in mopidy.conf.

```yaml
version: "2.4"

services:
  icecast:
    container_name: icecast
    image: rfabri/icecast:amd64-latest
    restart: unless-stopped
    environment:
      - ICECAST_SOURCE_PASSWORD=mopidy
      - ICECAST_ADMIN_PASSWORD=mopidy
      - ICECAST_PASSWORD=mopidy
      - ICECAST_RELAY_PASSWORD=mopidy
    ports:
      - "0.0.0.0:8000:8000"

  mopidy-icecast:
    container_name: mopidy-icecast
    image: rfabri/mopidy:amd64-latest
    restart: unless-stopped

    ports:
      - "0.0.0.0:6601:6600"
      - "0.0.0.0:6681:6680"
    depends_on:
      - icecast
    volumes:
      - /volume1/Docker/mopidy-icesnap/mopidy/mopidy-icecast.conf:/mopidy/mopidy.conf # <- here
      - /volume1/Docker/mopidy-icesnap/:/mopidy
      - /volume1/Music:/mopidy/media:ro
```

But the `Mopidy-Spotify` extension is not working.

**Comment by RoggerFabri** [comment]
I'm glad you got it working by changing the `mopidy.conf` path, I'll update the docker-compose examples to reflect that too.

Yes, Mopidy-Spotify is [disabled ](https://github.com/RoggerFabri/mopidy-icesnap/blob/master/mopidy/Dockerfile#L18)on the current `Dockerfile`. The reason for that is that `libspotify` was retired and because of that Mopidy-Spotify no longer works, there's an issue [here](https://github.com/mopidy/mopidy-spotify/issues/110) tracking that. It is a work in progress and on the issue I have linked you will find some workarounds, I have seen people getting it to work by compiling the ` gst-plugin-spotify` locally. However, Mopidy-Spotify haven't released a version that contains that fix yet, and for that reason I have removed it from this docker image.
